### PR TITLE
[Ruby] fix ruby test, update error message

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- fix ruby test
- update error message

cc @cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

### CODE REVIEW

File: samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb [hunk 1]
```diff
@@ -156,7 +156,7 @@
     end
 
     it 'fails for invalid collection format' do
-      expect(proc { api_client.build_collection_param(param, :INVALID) }).to raise_error(RuntimeError, 'unknown collection format: :INVALID')
+      expect(api_client.build_collection_param(param, :INVALID)).to raise_error(RuntimeError, 'unknown collection format: :INVALID')
```
Comment (autopp) [inline samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb]: It should be a block:
```ruby
expect { api_client.build_collection_param(param, :INVALID) }.to
```

File: samples/openapi3/client/petstore/ruby/lib/petstore/api_client.rb [hunk 2]
```diff
@@ -379,7 +379,7 @@ def build_collection_param(param, collection_format)
         # return the array directly as typhoeus will handle it as expected
         param
       else
-        fail "unknown collection format: #{collection_format.inspect}"
+        fail "unknown collection format: #{collection_format}"
```
Comment (autopp) [inline samples/openapi3/client/petstore/ruby/lib/petstore/api_client.rb]: I think it's better not to fix it here.
For example, nothing is output when `collection_format` is nil.

File: samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb (hunk 1 above)
Comment (wing328) [reply samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb]: I simply followed lines 155, 151, 147, etc.

Not sure why `proc` was used in the first place but looks like it doesn't work with the newer version of the dependency or ruby version (updated in the CI container)

File: samples/openapi3/client/petstore/ruby/lib/petstore/api_client.rb (hunk 2 above)
Comment (wing328) [reply samples/openapi3/client/petstore/ruby/lib/petstore/api_client.rb]: Sure. Will revert the change as there should be a better fix

File: samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb (hunk 1 above)
Comment (autopp) [reply samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb]: If wee use `raise_error`, wee need to pass a block to capture the exception.
(And 155, 151, 147, etc don't capture the exception)

> it doesn't work with the newer version of the dependency or ruby version (updated in the CI container)

Hmm... I don't know the details, but the behavior of RSpec may have changed.
In any case, it worked as expected with `expect { api_client.build_collection_param(param, :INVALID) }.to` in my environment. Can you give it a try?

Comment (wing328) [reply samples/openapi3/client/petstore/ruby/spec/api_client_spec.rb]: Yup, you're right that the behavior has changed somehow. Just tried it locally and using block fixes it 👍

**Comment by josh08h** [comment]
Is there a timeframe on this being released and deployed into the `latest` image?

**Comment by wing328** [comment]
@josh08h It's already included in the v4.1.3 release: https://twitter.com/oas_generator/status/1180123829626003456

**Pull request merged**